jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/JibPluginTest.java

```diff
@@ -368,6 +368,28 @@ public void testLazyEvalForLabels() {
             "labels contain values [firstkey:updated-first-label, secondKey:updated-second-label]");
   }
 
+  @Test
+  public void testLazyEvalForExtraDirectories() {
+    BuildResult checkExtraDirectories =
+        testProject.build("check-extra-directories", "-Djib.console=plain");
+    assertThat(checkExtraDirectories.getOutput()).contains("[/updated:755]");
+    assertThat(checkExtraDirectories.getOutput()).contains("updated-custom-extra-dir");
+  }
+
+  @Test
+  public void testLazyEvalForExtraDirectories_IndividualPaths() {
+    BuildResult checkExtraDirectories =
+        testProject.build(
+            "check-extra-directories", "-b=build-extra-dirs.gradle", "-Djib.console=plain");
+    assertThat(checkExtraDirectories.getOutput()).contains("updated-custom-extra-dir");
```

Comment:
How about we do this instead?
```suggestion
    assertThat(checkExtraDirectories.getOutput()).contains("extraDirectories (from): [/updated-custom-into-dir]");
```

Reply:
The blocker here was that the path will be resolved relative to the [TestProject](https://github.com/GoogleContainerTools/jib/blob/0e731e241609684c0658337115e4370b1a93caae/jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/JibPluginTest.java#L67) temp folder. Relative path to test project resources is used because there is currently a check (#3550) that the directory must actually exist, hence the unused file commit above). 

This output filepath will therefore look something like `/<path-to-tmp>/junit<####>/junit<####>/src/main/updated-custom-extra-dir` which becomes hard to assert against. Not sure if there is a more elegant workaround I am missing here!

Reply:
Ah I see! This can be tricky especially if the root directory of the `TestProject` keeps changing. We have a couple of examples in our tests that reference the `TestProject` and use `.resolve` to concatenate on only the relevant child directories which might be helpful: https://github.com/GoogleContainerTools/jib/blob/3e694e7c7fcb3f56c333359ada9fb40450cf771c/jib-cli/src/integration-test/java/com/google/cloud/tools/jib/cli/WarCommandTest.java#L84. 

Additionally, unrelated but in order to reference resource files under `src/test/resources`, this command tends comes in handy!: `Paths.get(Resources.getResource(directory).toURI())`. Here is another [example](https://github.com/GoogleContainerTools/jib/blob/5e9a65f48a1631f32afbbe1bf5b574f38bb52d9c/jib-core/src/test/java/com/google/cloud/tools/jib/api/JavaContainerBuilderTest.java#L39) in our test code.

Reply:
Updated - thank you for the pointers here!